src/main/java/macrobase/analysis/BatchAnalyzer.java

```diff
@@ -29,9 +35,41 @@ public BatchAnalyzer(MacroBaseConf conf) throws ConfigurationException {
         super(conf);
         conf.sanityCheckBatch();
     }
+    
+    public void contextualAnalyze() throws SQLException, IOException, ConfigurationException{
+    	OutlierDetector detector = constructDetector(randomSeed);
+    	
+    	
+    	DataLoader loader = constructLoader();
+    	DatumEncoder encoder = new DatumEncoder();
+    	List<Datum> data = loader.getData(encoder,
+    			contextualDiscreteAttributes,contextualDoubleAttributes);
+    	
+    	
+    	
+    	ContextualOutlierDetector contextualDetector = new ContextualOutlierDetector(detector,
+    			contextualDiscreteAttributes,contextualDoubleAttributes,contextualDenseContextTau,contextualNumIntervals);
+    	
+    	contextualDetector.searchContextualOutliers(data, zScore);
+    	Map<Context,OutlierDetector.BatchResult> context2Outliers = contextualDetector.getContextualOutliers();
+        for(Context context: context2Outliers.keySet()){
+        	System.out.println("Context: " + context.print(encoder));
+        	System.out.println("Number of Inliers: " + context2Outliers.get(context).getInliers().size());
+        	System.out.println("Number of Outliers: " + context2Outliers.get(context).getOutliers().size());
+      
+        }
+    	//explain the contextual outliers
+    }
 
     public AnalysisResult analyze() throws SQLException, IOException, ConfigurationException {
-        DatumEncoder encoder = new DatumEncoder();
+        
+    	if(contextualSwitch){
```

Comment:
Since the contextual analysis will include the global context, so I felt it can be removed.

 But right now, the contextual analysis does not have explanations. So I am keeping the original analysis.